ref.pl

```diff
@@ -687,6 +687,33 @@
     appear on the screen, where the new value can be entered. Press
     enter to commit the change, or Esc to cancel.
 
+<h3>Automatic Constraints</h3>
+
+<p>
+    There is an option in the configuration screens labled "enable
+    automatic line constraint". If this option is checked and a line
+    being drawn is nearly horizontal, a horizontal contraint will
+    appear over the middle of the line and will be applied when the
+    second point of the line is clicked in place. Similarly when a
+    nearly vertical line is being drawn a "V" constraint will appear
+    automatically. If the line is not close enough to horizontal or
+    vertical, no constraint will be added.
+    
+<p>
+    If an automatic constraint is redundant or would result in the
+    sketch being over-constrained, it will not be created even if
+    it is shown during line placement. Similarly if a distance or
+    diameter constraint is added that would over-constrain the
+    sketch, it will be created as a reference (REF).
+
+<p>
+    Another option in the configuration screen is "edit newly added
+    dimensions". With this option enabled, any time a new distance
+    constraint is added the input box will automatically appear so
```

Comment:
With this option enabled, when a new Distance, Length Ratio or Length Difference